git_connector.py

```diff
@@ -741,65 +756,87 @@ def _test_asset_connectivity(self, param):
         action_result = ActionResult()
         self._set_repo_attributes(param=param)
         self.save_progress(consts.GIT_CONNECTION_TEST_MSG)
-        self.save_progress('Configured repo URI: {}'.format(self.repo_uri))
+        self.save_progress("Configured repo URI: {}".format(self.repo_uri))
 
         remote_refs = []
         g = git.cmd.Git()
 
         # ls_remote function call the git command ls-remote
         try:
-            repo_details = g.ls_remote(self.modified_repo_uri).split('\n')
+            repo_details = g.ls_remote(self.modified_repo_uri).split("\n")
         except Exception:
             if not self.repo_uri:
                 self.save_progress(
                     "You haven't added a repo URI to test connectivity to.  Only 'clone repo' with a provided repo URL will work!"
                 )
             else:
-                self.save_progress('Error while calling configured URI')
+                self.save_progress("Error while calling configured URI")
             if self.ssh:
-                self.save_progress('Do you still need to run the configure_ssh action?')
+                self.save_progress("Do you still need to run the configure_ssh action?")
             self.set_status(phantom.APP_ERROR, consts.GIT_TEST_CONNECTIVITY_FAIL)
             return action_result.get_status()
 
         # create a list of refs
         for ref in repo_details:
             # if repo is empty
             if ref:
-                remote_refs.append(ref.split('\t')[1])
+                remote_refs.append(ref.split("\t")[1])
 
         # if branch name is not in ref_list it is invalid
         for ref in remote_refs:
-            if self.branch_name == ref.split('/')[-1]:
+            if self.branch_name == ref.split("/")[-1]:
                 break
         else:
-            self.save_progress('Invalid branch name')
+            self.save_progress("Invalid branch name")
             self.set_status(phantom.APP_ERROR, consts.GIT_TEST_CONNECTIVITY_FAIL)
             return action_result.get_status()
 
         self.set_status_save_progress(phantom.APP_SUCCESS, consts.GIT_TEST_CONNECTIVITY_SUCCESS)
         return action_result.get_status()
 
+    def _on_poll(self, param):
+        """This function will attempt to either perform a pull or clone depending on results of git status on a schedule
+            _on_poll will make use of asset configs
+        :param param: dictionary of input parameters
+        """
```

Comment:
This isn't really the intended use case for `on poll`. `on poll` actions should create containers and/or artifacts in SOAR. It looks to me like we're trying to use the built-in scheduling capability of `on poll` to use the `git` connector to periodically update repositories automatically. Is that correct?

Reply:
Yes. The on-poll action here is not creating any containers or artifacts, just using the its scheduling capability for cloning or pulling the git repo - ie for synchronizing.

This is not the first time when on-poll is not used in the intended way. In slack, on_poll is used to trigger the slack bot and nothing else.